src/vma/sock/sockinfo.cpp

```diff
@@ -926,6 +926,14 @@ bool sockinfo::attach_as_uc_receiver(role_t role, bool skip_rules /* = false */)
 			}
 		}
 	}
+
+	if (m_rx_ring_map.size() == 1) {
+		rx_ring_map_t::iterator rx_ring_iter = m_rx_ring_map.begin();
+		m_p_rx_ring = rx_ring_iter->first;
+	} else {
+		si_logdbg("ring map size: %d", m_rx_ring_map.size());
+	}
+
```

Comment:
I think that it should be done in case ret == true only.
Please consider consistence changes for TCP flow.

Use following code as an example:
```
	bool success = attach_as_uc_receiver((role_t)NULL, true);
	if (!success) {
		setPassthrough();
		unlock_tcp_con();
		si_tcp_logdbg("non offloaded socket --> connect only via OS");
		return -1;
	}
```
See 
https://github.com/Mellanox/libvma/blob/experimental_256B/src/vma/sock/sockinfo_tcp.cpp#L1870-L1882
https://github.com/Mellanox/libvma/blob/experimental_256B/src/vma/sock/sockinfo_tcp.cpp#L2173-L2195
https://github.com/Mellanox/libvma/blob/experimental_256B/src/vma/sock/sockinfo_tcp.cpp#L2494-L2500